internal/core/src/index/Utils.cpp

```diff
@@ -68,6 +68,10 @@ unsupported_index_combinations() {
     static std::vector<std::tuple<IndexType, MetricType>> ret{
         std::make_tuple(knowhere::IndexEnum::INDEX_FAISS_BIN_IVFFLAT,
                         knowhere::metric::L2),
+        std::make_tuple(knowhere::IndexEnum::INDEX_SPARSE_INVERTED_INDEX,
```

Comment:
Don't quite understand.
Do we support cosine metrics?

Reply:
No, IP is the only supported metric. Updated.

Reply:
and bin ivf index don't support IP, cosine as well.

I think this check is meaningless.... what about INDEX_SPARSE_INVERTED_INDEX plus hamming

Reply:
agreed.. this check is never completed since the beginning and we always rely on checks elsewhere to guarantee the precondition. Since this check originally has only 1 pair, I'll see if I can remove it in a later PR.

sparse index has no hamming distance support.